lighthouse-core/computed/page-dependency-graph.js

```diff
@@ -186,15 +186,25 @@ class PageDependencyGraph {
    * @param {Array<CPUNode>} cpuNodes
    */
   static linkCPUNodes(rootNode, networkNodeOutput, cpuNodes) {
+    /** @type {Set<LH.Crdp.Network.ResourceType>} */
+    const linkableResourceTypes = new Set([
+      NetworkRequest.TYPES.XHR, NetworkRequest.TYPES.Fetch, NetworkRequest.TYPES.Script,
+    ]);
+
     /** @param {CPUNode} cpuNode @param {string} reqId */
     function addDependentNetworkRequest(cpuNode, reqId) {
       const networkNode = networkNodeOutput.idToNodeMap.get(reqId);
       if (!networkNode ||
-          // Ignore all non-XHRs
-          networkNode.record.resourceType !== NetworkRequest.TYPES.XHR ||
           // Ignore all network nodes that started before this CPU task started
           // A network request that started earlier could not possibly have been started by this task
           networkNode.startTime <= cpuNode.startTime) return;
+      const resourceType =
+        networkNode.record.resourceType || networkNode.record.redirectedResourceType;
+      if (typeof resourceType !== 'undefined' && !linkableResourceTypes.has(resourceType)) {
```

Comment:
if this is just for tsc, could change the declared type of the set to include undefined. I scratched my head for a bit on why the `typeof` check was there :)

Reply:
Done